x-pack/filebeat/module/mysqlenterprise/audit/ingest/pipeline.yml

```diff
@@ -0,0 +1,158 @@
+- append:
+    field: event.type
+    value: end
+    if: ctx?.mysqlenterprise?.audit?.event == 'disconnect'
+- set:
+    field: event.outcome
+    value: success
+    if: ctx?.mysqlenterprise?.audit?.connection_data?.status != null && ctx?.mysqlenterprise?.audit?.connection_data?.status == 0 || ctx?.mysqlenterprise?.audit?.general_data?.status != null && ctx?.mysqlenterprise?.audit?.general_data?.status == 0
+- set:
+    field: event.outcome
+    value: failure
+    if: ctx?.mysqlenterprise?.audit?.connection_data?.status != null && ctx?.mysqlenterprise?.audit?.connection_data?.status > 0 || ctx?.mysqlenterprise?.audit?.general_data?.status != null && ctx?.mysqlenterprise?.audit?.general_data?.status > 0
+- set:
+    field: event.outcome
+    value: unknown
+    if: ctx?.event?.outcome == null
+- set:
+    field: event.action
+    value: mysql-{{ mysqlenterprise.audit.event }}
+    if: ctx?.mysqlenterprise?.audit?.event != null
+- script:
+    lang: painless
+    if: ctx?.mysqlenterprise?.audit != null
+    source: |
+      void handleMap(Map map) {
+        for (def x : map.values()) {
+          if (x instanceof Map) {
+              handleMap(x);
+          } else if (x instanceof List) {
+              handleList(x);
+          }
+        }
+        map.values().removeIf(v -> v instanceof String && v.isEmpty() == true);
+      }
+      void handleList(List list) {
+        for (def x : list) {
+            if (x instanceof Map) {
+                handleMap(x);
+            } else if (x instanceof List) {
+                handleList(x);
+            }
+        }
+      }
+      handleMap(ctx);
+- rename:
+    field: mysqlenterprise.audit.account.user
+    target_field: server.user.name
+    ignore_missing: true
+- rename:
+    field: mysqlenterprise.audit.account.host
+    target_field: client.domain
+    ignore_missing: true
+- rename:
+    field: mysqlenterprise.audit.login.os
+    target_field: client.user.name
+    ignore_missing: true
+- rename:
+    field: mysqlenterprise.audit.login.ip
+    target_field: client.ip
+    ignore_missing: true
+- rename:
+    field: mysqlenterprise.audit.startup_data.os_version
+    target_field: host.os.full
+    ignore_missing: true
+- rename:
+    field: mysqlenterprise.audit.startup_data.mysql_version
+    target_field: service.version
+    ignore_missing: true
+- rename:
+    field: mysqlenterprise.audit.startup_data.server_id
+    target_field: service.id
+    ignore_missing: true
+- rename:
+    field: mysqlenterprise.audit.startup_data.args
+    target_field: process.args
+    ignore_missing: true
+- set:
+    field: process.name
+    value: mysqld
+- join:
+    field: process.args
+    target_field: process.command_line
+    separator: " "
+    ignore_failure: true
+- script:
+    lang: painless
+    if: ctx?.process?.args != null
+    source: |
+      ctx.process.args_count = ctx.process.args.length - 1;
+      ctx.process.executable = ctx.process.args[0];
```

Comment:
Being extra paranoid here, I think this will cause an exception when process.args is an empty list (non-null, but zero length).